src/sdk/map_client.js

```diff
@@ -557,6 +560,43 @@ class MapClient {
             }));
     }
 
+    async move_blocks_to_storage_class() {
+        const blocks = [];
+        const target_storage_class = this.move_to_tier.storage_class;
+
+        for (const chunk of this.chunks) {
+            // There is no straight forward way to determine that the block indeed needs to
+            // be moved to the storage class because map_builder updates the chunk's tier prior
+            // to invoking map_client due to which we lose the information about the chunk's
+            // previous tier.
+            for (const frag of chunk.frags) {
+                for (const block of frag.blocks) {
+                    blocks.push(block);
+                }
+            }
+        }
+
+        if (blocks.length === 0) return;
+
+        const blocks_by_agent = _.groupBy(blocks, 'address');
+        await P.map(_.keys(blocks_by_agent), async agent_address => {
+            const blocks_for_agent = blocks_by_agent[agent_address];
+            const block_ids = _.map(blocks_for_agent, block => block._id?.toString());
```

Comment:
is there a case where _id is undefined? is that valid? if so we might want to filter it out and not leave those undefined values in the array.